ring_doorbell/auth.py

```diff
@@ -1,35 +1,71 @@
 # coding: utf-8
 # vim:sw=4:ts=4:et:
 """Python Ring Auth Class."""
+try:
+    from urllib.parse import urlencode
+except ImportError:
+    from urllib import urlencode
+
 from requests_oauthlib import OAuth2Session
 from oauthlib.oauth2 import (
     LegacyApplicationClient, TokenExpiredError,
     MissingTokenError)
-from ring_doorbell.const import OAuth
+from ring_doorbell.const import (OAuth, CACHE_FILE, CACHE_ATTRS,
+                                 API_VERSION, TIMEOUT, RETRY_TOKEN)
+from ring_doorbell.utils import _exists_cache, _save_cache, _read_cache
 
 
 class Auth:
     """A Python Auth class for Ring"""
     def __init__(self,
                  token=None,
```

Comment:
We're having two types of caches happening here.

- `token` + `token_updater` is the current token cache plus a callback to store the cache when it has changed.
- `cache_file` is a path to a file that data is written to. Including the token.

First question is probably best answered by @tchellomello, why do we cache some data and not just fetch it freshly from Ring API?

If we decide to keep the cache around, we should merge `cache_file` into `token_updater`. The `token_updater` approach is nice, because it removes all caching logic from the library. It just calles a method and it's up to the consumer of the lib to decide how to store it.